Updating IBM Watson SDK

rutkai opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
None

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
* Updated WatsonTTS SDK: version bump from 5.1.0 -> [5.2.2](https://github.com/watson-developer-cloud/python-sdk/blob/master/CHANGELOG.md#522-2021-07-06)

https://github.com/watson-developer-cloud/python-sdk/compare/v5.1.0...v5.2.2

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [X] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes 
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [X] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

rutkai committed:
Updating IBM Watson SDK, fixing invalid function call

MartinHjelmare commented:
Please complete the relevant parts of the checklist.

- [ ] New or updated dependencies have been added to requirements_all.txt.
Updated by running python3 -m script.gen_requirements_all.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

MartinHjelmare commented on homeassistant/components/watson_tts/tts.py:
```diff
@@ -191,7 +191,7 @@ def supported_options(self):
         """Request TTS file from Watson TTS."""
         response = self.service.synthesize(
-            message, accept=self.output_format, voice=self.default_voice
+            text=message, accept=self.output_format, voice=self.default_voice
```
Looks like https://github.com/home-assistant/core/pull/54916 has the same fix. Unless the library bump is needed for the fix we prefer to separate fixes from other types of changes into different PRs.

GuyKh replied:
Sorry. I missed @rutkai fix. Should I close my PR and we'll proceed with this one?

MartinHjelmare replied:
It's best to keep the fix separate from the library bump unless the fix requires the library bump. So I suggest we remove the fix here after confirming the above.

rutkai replied:
It's fine to merge your change first, then merging mine would bump the version only - or I'm fine closing my PR. No strings attached ;)

MartinHjelmare replied:
I've merged the other PR. Please rebase here on latest dev branch to nly include the library version bump.

rutkai commented:
> Please complete the relevant parts of the checklist.
> 
> * [ ]  New or updated dependencies have been added to requirements_all.txt.
>   Updated by running python3 -m script.gen_requirements_all.
> * [ ]  For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

I updated `requirements_all.txt` manually, that should do it, right?
I checked both checkboxes and updated the description.

MartinHjelmare commented:
Thanks, please paste a link to a changelog or github commit compare range for the library version bump in the PR description.

rutkai commented:
Updated the branch and the PR description. This should be good to go. :)

MartinHjelmare commented:
I don't see a link to a changelog or a link to a github commit compare range for the library version bump. Can you add one? Thanks!

rutkai commented:
Updated the description with a link to the [changelog](https://github.com/watson-developer-cloud/python-sdk/blob/master/CHANGELOG.md#522-2021-07-06).

frenck reviewed (approved):
LGTM, Thanks @rutkai 👍

MartinHjelmare reviewed (approved):
I'll approve this upgrade, even though ibm-watson pins websocket-client to an exact version which we normally don't approve when upgrading requirements. It is not worse than before since it pinned websocket-client in 5.1.0 to 0.48.0, while now it pins it to 1.1.0
